### ISSUE TITLE
Incorrectly marking class variable as unused when destructured from this.

### ISSUE THREAD

**Issue by zerovox** [palantir/tslint#850]
```ts
class foo {
  bar () {
    let {a} = this;
    a();
  }
 
 private a = () => {};
}
```

TSLint will error with 'foo.ts[7, 11]: unused variable: 'a''.

**Comment by JKillian** [comment]
Hi @zerovox, this seems like a TSLint bug, but it's actually not something we can do much about. The `this` in `bar(){...}` could refer to something other than the current instance of the class depending on how it is called, thus it can't be counted as a usage of the class member `a` for sure.
